dbt/include/databricks/macros/adapters.sql

```diff
@@ -105,16 +115,234 @@
 {% endmacro %}
 
 {% macro databricks__persist_constraints(relation, model) %}
-  {# Model contracts are not currently supported. #}
+
+  {% set delta_format = config.get('file_format', 'delta') == 'delta' %}
+
   {%- set contract_config = config.get('contract') -%}
-  {% if contract_config and contract_config.enforced %}
-    {{ exceptions.raise_compiler_error('Model contracts are not currently supported.') }}
+  {% set has_model_contract = contract_config and contract_config.enforced %}
+
+  {% set has_databricks_constraints = config.get('persist_constraints', False) %}
+
+  {% if (has_model_contract or has_databricks_constraints) and delta_format and not is_incremental() %}
+    {% do alter_table_add_constraints(relation, model) %}
+    {% do alter_column_set_constraints(relation, model) %}
+  {% endif %}
+{% endmacro %}
+
+{% macro alter_table_add_constraints(relation, constraints) %}
+  {{ return(adapter.dispatch('alter_table_add_constraints', 'dbt')(relation, constraints)) }}
+{% endmacro %}
+
+{% macro databricks__alter_table_add_constraints(relation, model) %}
+    {% set constraints = get_model_constraints(model) %}
+
+    {% set statements = get_constraints_sql(relation, constraints, model) %}
+    {% for stmt in statements %}
+      {% call statement() %}
+        {{ stmt }}
+      {% endcall %}
+    {% endfor %}
+
+{% endmacro %}
+
+{% macro get_model_constraints(model) %}
+  {% set constraints = model.get('constraints', []) %}
+  {% set has_databricks_constraints = config.get('persist_constraints', False) and config.get('file_format', 'delta') == 'delta' %}
```

Comment:
how would the execution even get to here? Do we return a nice error when users set constraints to tables that are not in delta format?

Reply:
I broke this out into its own macro for testing.
If there is either a model contract or databricks constraints this code will be executed.  Where in the model the constraints are located depends on which.

Reply:
the check for delta needs to happen as early as possible as we only support contraints in delta